profiler/src/main/java/com/splunk/opentelemetry/profiler/context/StackDescriptorLineParser.java

```diff
@@ -28,30 +30,31 @@
    * The first line is a meta/descriptor that has information about the following call stack. This
    * method parses out the thread id, which is the second field (space separated).
    */
-  long parseThreadId(String descriptorLine) {
+  long parseThreadId(ThreadDumpRegion stack) {
```

Comment:
Nit: another idea: `ThreadDumpRegion` could implement `CharSequence`, this class would need almost no changes then. And you could use simple strings to test it

Reply:
That would require managing a relative vs absolute offset within `ThreadDumpRegion` (externally only relative offset is used, internally needs to mapped to absolute, and vice versa). While it might simplify the logic here, it would add unnecessary much complexity to `ThreadDumpRegion` itself.